cmd/thanos/bucket.go

```diff
@@ -80,6 +80,9 @@ func registerBucketVerify(m map[string]setupFunc, root *kingpin.CmdClause, name
 		Short('i').Default(verifier.IndexIssueID, verifier.OverlappedBlocksIssueID).Strings()
 	idWhitelist := cmd.Flag("id-whitelist", "Block IDs to verify (and optionally repair) only. "+
 		"If none is specified, all blocks will be verified. Repeated field").Strings()
+	deleteDelay := modelDuration(cmd.Flag("delete-delay", "Time before a block marked for deletion is deleted from bucket. "+
+		"If delete-delay is non zero, blocks are marked for deletion and compactor component is required to delete blocks from source bucket.").
```

Comment:
Nice! I would even add more: `If delete is 0 blocks will be deleted straight away. Use this if you want to get rid of or move the block immediately.  Note that this might mean artifacts like query failures for store gateway that still have this block loaded or compactor ignoring delete because it's compacting this block now.`

Reply:
made this change. Agree that adding an explanation of what will happen if its 0 is required.